Flaky integration tests with Zeebe 1.0.0-rc3

**Describe the bug**
After upgrading to Zeebe 1.0.0-rc3 in GitHub Actions, the tests have started to fail. 

From the looks, it seems to be caused by `zeebe_worker._watcher_thread.is_alive()` not returning `False` - probably because the watcher thread hasn't stopped. 

**To Reproduce**
1) Checkout https://github.com/kbakk/pyzeebe/commit/3a355aa39f9665102f00f57d444617528ad97b16. 
2) Run a local broker with `camunda/zeebe:1.0.0-rc3`. 
3) Run the integration tests (`pytest tests/integration`) a few times. 

**Expected behavior**
Tests succeeding. 

**`pytest` error**

<details>
<summary>output</summary>

```console
❯ pytest tests/integration
================================================================================================================================================= test session starts ==================================================================================================================================================
platform darwin -- Python 3.8.9, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /Users/krisb/Code/fork/pyzeebe
plugins: grpc-0.8.0, mock-3.5.1
collected 4 items                                                                                                                                                                                                                                                                                                      

tests/integration/integration_test.py ....Exception in thread ZeebeWorker-Job-test:
Traceback (most recent call last):
  File "/Users/krisb/Code/fork/pyzeebe/pyzeebe/grpc_internals/zeebe_job_adapter.py", line 51, in complete_job
    return self._gateway_stub.CompleteJob(CompleteJobRequest(jobKey=job_key, variables=json.dumps(variables)))
  File "/Users/krisb/.local/share/virtualenvs/pyzeebe-oIjZHgFZ/lib/python3.8/site-packages/grpc/_channel.py", line 946, in __call__
    return _end_unary_response_blocking(state, call, False, None)
  File "/Users/krisb/.local/share/virtualenvs/pyzeebe-oIjZHgFZ/lib/python3.8/site-packages/grpc/_channel.py", line 849, in _end_unary_response_blocking
    raise _InactiveRpcError(state)
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
        status = StatusCode.NOT_FOUND
        details = "Command 'COMPLETE' rejected with code 'NOT_FOUND': Expected to complete job with key '2251799813685370', but no such job was found"
        debug_error_string = "{"created":"@1620404894.359087000","description":"Error received from peer ipv6:[::1]:26500","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Command 'COMPLETE' rejected with code 'NOT_FOUND': Expected to complete job with key '2251799813685370', but no such job was found","grpc_status":5}"
>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/krisb/.pyenv/versions/3.8.9/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/Users/krisb/.pyenv/versions/3.8.9/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/krisb/Code/fork/pyzeebe/pyzeebe/task/task_builder.py", line 34, in job_handler
    job.set_success_status()
  File "/Users/krisb/Code/fork/pyzeebe/pyzeebe/job/job.py", line 40, in set_success_status
    self.zeebe_adapter.complete_job(
  File "/Users/krisb/Code/fork/pyzeebe/pyzeebe/grpc_internals/zeebe_job_adapter.py", line 54, in complete_job
    raise JobNotFoundError(job_key=job_key)
pyzeebe.errors.job_errors.JobNotFoundError: Job 2251799813685370 not found
E                                                                                                                                                                                                                                                                      [100%]

======================================================================================================================================================== ERRORS ========================================================================================================================================================
_______________________________________________________________________________________________________________________________________ ERROR at teardown of test_cancel_process _______________________________________________________________________________________________________________________________________

zeebe_worker = <pyzeebe.worker.worker.ZeebeWorker object at 0x107ddb460>, zeebe_client = <pyzeebe.client.client.ZeebeClient object at 0x107d99970>

    @pytest.fixture(scope="module", autouse=True)
    def setup(zeebe_worker, zeebe_client):
        zeebe_worker.work(watch=True)
    
        try:
            integration_tests_path = os.path.join("tests", "integration")
            zeebe_client.deploy_process(
                os.path.join(integration_tests_path, "test.bpmn")
            )
        except FileNotFoundError:
            zeebe_client.deploy_process("test.bpmn")
    
        yield
        zeebe_worker.stop(wait=True)
>       assert not zeebe_worker._watcher_thread.is_alive()
E       assert not True
E        +  where True = <bound method Thread.is_alive of <Thread(ZeebeWorker-Watch, started 123145540276224)>>()
E        +    where <bound method Thread.is_alive of <Thread(ZeebeWorker-Watch, started 123145540276224)>> = <Thread(ZeebeWorker-Watch, started 123145540276224)>.is_alive
E        +      where <Thread(ZeebeWorker-Watch, started 123145540276224)> = <pyzeebe.worker.worker.ZeebeWorker object at 0x107ddb460>._watcher_thread

tests/integration/integration_test.py:47: AssertionError
=============================================================================================================================================== short test summary info ================================================================================================================================================
ERROR tests/integration/integration_test.py::test_cancel_process - assert not True
============================================================================================================================================== 4 passed, 1 error in 0.74s ==============================================================================================================================================

```

</details>

See also GitHub Actions run: https://github.com/kbakk/pyzeebe/runs/2529153752?check_suite_focus=true#step:7:73

--- a/pyzeebe/worker/worker.py
+++ b/pyzeebe/worker/worker.py
@@ -95,6 +95,8 @@ class ZeebeWorker(ZeebeTaskRouter):
         self.stop_event.set()
         if wait:
             self._join_task_threads()
+            if self._watcher_thread:
+                self._watcher_thread.join()
 
     def _join_task_threads(self) -> None:
         logger.debug("Waiting for threads to join")

